cc_core.c

```diff
@@ -2162,9 +2164,32 @@ void collect_local(void)
 			else if(RISCV64 == Architecture) a->depth = function->locals->depth - register_size;
 		}
 
+		function->locals = a;
+
+		emit_out("# Defining local ");
+		emit_out(global_token->s);
+		emit_out("\n");
+
+		global_token = global_token->next;
+		require(NULL != global_token, "incomplete local missing name\n");
+
+		a->array_modifier = 1;
+		if(match("[", global_token->s))
+		{
+			maybe_bootstrap_error("array on the stack");
+
+			global_token = global_token->next;
+			require(NULL != global_token, "incomplete local array\n");
+
+			a->array_modifier = strtoint(global_token->s);
+
+			global_token = global_token->next;
+			require_match("ERROR in collect_local\nMissing ] after local array size\n", "]");
```

Comment:
this would work but only for simple cases... In C size has to be known at compile time but something like i.e. `int  arr[5+5];` is valid but would fail here.

But perhaps that's good enough for now since it will catch the most common use case... Though worth clarifying that in changelog/release notes.

Reply:
Hmm, I wonder if we could use `expression()` here? I guess on the stack it would basically be VLA...